File: sales_analysis.R
# sales_analysis.R

# Load required libraries
library(readxl)
library(ggplot2)

# Step 1: Read data from Excel file
data_file <- "sales_data.xlsx"
df <- read_excel(data_file)

# Step 2: Data Preprocessing
# Convert 'Date' column to Date format
df$Date <- as.Date(df$Date)

# Step 3: Data Analysis
# Calculate total sales
total_sales <- sum(df$Sales)
cat(sprintf("Total Sales: $%0.2f\n", total_sales))

# Calculate sales by product
sales_by_product <- aggregate(Sales ~ Product, data = df, sum)
cat("\nSales by Product:\n")
print(sales_by_product)

# Step 4: Data Visualization
# Sales by Product Bar Chart
ggplot(sales_by_product, aes(x = Product, y = Sales, fill = Product)) +
  geom_bar(stat = "identity") +
  ggtitle("Sales by Product") +
  xlab("Product") +
  ylab("Total Sales") +
  theme_minimal()

# Save the plot
ggsave("sales_by_product.png")

# Sales Over Time Line Chart
ggplot(df, aes(x = Date, y = Sales, color = Product)) +
  geom_line() +
  ggtitle("Sales Over Time by Product") +
  xlab("Date") +
  ylab("Sales") +
  theme_minimal()

# Save the plot
ggsave("sales_over_time.png")

# Step 5: Export Results to Excel
# Install the writexl package if not installed
# install.packages("writexl")
library(writexl)

write_xlsx(list("Sales by Product" = sales_by_product), "sales_analysis_results_R.xlsx")

cat("\nAnalysis results have been saved to 'sales_analysis_results_R.xlsx'\n")

File: sales_analysis.py
# sales_analysis.py

import pandas as pd
import matplotlib.pyplot as plt
import os

# Set your folder path
folder_path = r'C:\Users\Janay\OneDrive\Desktop\sales analysis'

# Step 1: Read data from Excel file
data_file = os.path.join(folder_path, 'sales_data.xlsx')
df = pd.read_excel(data_file)

# Step 2: Data Preprocessing
df['Date'] = pd.to_datetime(df['Date'])

# Step 3: Data Analysis
total_sales = df['Sales'].sum()
print(f"Total Sales: ${total_sales:,.2f}")

sales_by_region = df.groupby('Region')['Sales'].sum()
print("\nSales by Region:")
print(sales_by_region)

df['Month'] = df['Date'].dt.to_period('M')
monthly_sales = df.groupby('Month')['Sales'].sum()

# Step 4: Data Visualization
# Sales by Region Bar Chart
plt.figure(figsize=(8, 6))
sales_by_region.plot(kind='bar', color='skyblue')
plt.title('Sales by Region')
plt.xlabel('Region')
plt.ylabel('Total Sales')
plt.tight_layout()
plt.savefig(os.path.join(folder_path, 'sales_by_region.png'))
plt.show()

# Monthly Sales Line Chart
plt.figure(figsize=(10, 6))
monthly_sales.plot(kind='line', marker='o', linestyle='-')
plt.title('Monthly Sales Over Time')
plt.xlabel('Month')
plt.ylabel('Sales')
plt.tight_layout()
plt.savefig(os.path.join(folder_path, 'monthly_sales.png'))
plt.show()

# Step 5: Export Results to Excel
output_file = os.path.join(folder_path, 'sales_analysis_results.xlsx')
with pd.ExcelWriter(output_file) as writer:
    sales_by_region.to_frame().to_excel(writer, sheet_name='Sales by Region')
    monthly_sales.to_frame().to_excel(writer, sheet_name='Monthly Sales')

print(f"\nAnalysis results have been saved to '{output_file}'")
